compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -1783,7 +1783,10 @@ trait Applications extends Compatibility {
     }
 
     def resultIsMethod(tp: Type): Boolean = tp.widen.stripPoly match
-      case tp: MethodType => tp.resultType.isInstanceOf[MethodType]
+      case tp: MethodType =>
+        tp.resultType match
+          case mt: MethodType => !mt.isContextualMethod
```

Comment:
That does not handle the case correctly where a context method type has a regular method type as a result. You can do this instead:
```scala
case tp: MethodType => tp.resultType.stripImplicit.isInstanceOf[MethodType]
```